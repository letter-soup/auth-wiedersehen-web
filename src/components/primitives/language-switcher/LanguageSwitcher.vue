<script setup lang="ts">
import { Select, SelectContent, SelectItem, SelectTrigger } from '@/components/ui/select'
import { Languages } from 'lucide-vue-next'
import { useI18n } from 'vue-i18n'

const { locale: selectedLocale, availableLocales } = useI18n()
</script>

<template>
  <Select v-model="selectedLocale">
    <SelectTrigger class="w-fit h-fit">
      <Languages :size="16" />
    </SelectTrigger>
    <SelectContent>
      <SelectItem v-for="locale in availableLocales" :key="locale" :value="locale">
        {{ $t(`language:${locale}`) }}
      </SelectItem>
    </SelectContent>
  </Select>
</template>

<style scoped></style>
