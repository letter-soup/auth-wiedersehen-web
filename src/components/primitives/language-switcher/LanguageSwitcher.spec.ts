import { describe, expect, it, vi } from 'vitest'
import { mount } from '@vue/test-utils'
import LanguageSwitcher from '@/components/primitives/language-switcher/LanguageSwitcher.vue'

vi.mock('lucide-vue-next', () => ({
  Languages: 'languages-icon',
  ChevronDown: 'chevron-down-icon',
  Check: 'check-icon',
}))

describe('LanguageSwitcher (snapshot)', () => {
  it('renders properly', async () => {
    const wrapper = mount(LanguageSwitcher)
    await wrapper.vm.$nextTick()

    expect(wrapper).toBeTruthy()
    expect(wrapper.html()).toMatchSnapshot()
  })
})

describe('LanguageSwitcher', () => {
  it('given locale:en-US when locale:ru-RU selected then updates locale', async () => {
    const wrapper = mount(LanguageSwitcher)
    await wrapper.vm.$nextTick()

    const select = wrapper.find('[data-testid=language-switcher]')
    expect(select.attributes()['data-selected-locale']).toBe('en-US')
    await select.setValue('ru-RU')
    expect(select.attributes()['data-selected-locale']).toBe('ru-RU')
  })
})
