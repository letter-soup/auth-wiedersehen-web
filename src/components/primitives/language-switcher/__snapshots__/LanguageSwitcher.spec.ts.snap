// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`LanguageSwitcher (snapshot) > renders properly 1`] = `
"<button role="combobox" type="button" aria-controls="radix-vue-select-content-v-0" aria-expanded="false" aria-required="false" aria-autocomplete="none" dir="ltr" data-state="closed" class="flex items-center justify-between rounded-md border border-input bg-background px-3 py-2 text-sm ring-offset-background placeholder:text-muted-foreground focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 disabled:cursor-not-allowed disabled:opacity-50 [&amp;>span]:truncate text-start w-fit h-fit">
  <languages-icon size="16"></languages-icon>
  <chevron-down-icon class="w-4 h-4 opacity-50 shrink-0" aria-hidden="true"></chevron-down-icon>
</button>
<!--teleport start-->
<!--teleport end-->
<select default-value="en-US" style="position: absolute; border: 0px; width: 1px; display: inline-block; height: 1px; padding: 0px; margin: -1px; overflow: hidden; clip: rect(0px, 0px, 0px, 0px); white-space: nowrap; word-wrap: normal;" data-testid="language-switcher" data-selected-locale="en-US" aria-hidden="true" tabindex="-1" value="en-US">
  <!---->
  <option value="en-US">language:en-US</option>
  <option value="ru-RU">language:ru-RU</option>
</select>"
`;
